docs(web, web-react): add `Collapse` alternative to `Checkbox`/`Toggle` consent examples (#2810)
lmc-eu/spirit-design-system: packages/web-react/src/components/Checkbox/README.md

## Consent with Details

> For consent scenarios where users need access to supplementary content — such as terms and conditions or a privacy

Reviewer: ```suggestion
For consent scenarios where users need access to supplementary content — such as terms and conditions
```

Author: I fixed it in the examples but I'm keeping it here as relevant.

policy — use the `details` prop to render trigger links below the label. The linked content can be revealed in a
[Modal][readme-modal] dialog or expanded inline with [Collapse][readme-collapse].

**Keep the `label` itself as short as possible at all times.** Move any supplementary explanation into `details`
instead. Screen readers read the full label out loud, so a long label is read in full every time. Short labels are
also easier for sighted users to scan.

### With Modal

```tsx
import React, { useState } from 'react';
import { Checkbox, Link, Modal, Text } from '@alma-oss/spirit-web-react';

const Example = () => {
  const [isTermsModalOpen, setTermsModalOpen] = useState(false);

  return (
    <>
      <Checkbox
        id="consent"

>         label="I agree to the terms and privacy policy"

Reviewer: ```suggestion
        label="I agree to the terms and conditions"
```

        isRequired
        details={
          <>
            <Text size="small" marginBottom="space-0">
              Please review our terms and conditions before you agree.
            </Text>

>             <Link elementType="button" color="inherit" underlined="always" onClick={() => setTermsModalOpen(true)}>

Reviewer: ```suggestion
            <Link elementType="button" color="inherit" underlined="always" onClick={() => setTermsModalOpen(true)} aria-controls="checkbox-terms-modal" aria-expanded={isTermsModalOpen}>
```

Reviewer: It would be useful to better document these aria attributes in the Modal/README.md as well.

              See full terms and conditions
            </Link>
          </>
        }
      />
      <Modal id="checkbox-terms-modal" isOpen={isTermsModalOpen} onClose={() => setTermsModalOpen(false)}>
        {/* Modal content */}
      </Modal>
    </>
  );
};
```

### With Collapse

```tsx
import React from 'react';
import { Checkbox, Collapse, Link, Text, useCollapse } from '@alma-oss/spirit-web-react';

const Example = () => {
  const { isOpen, toggleHandler } = useCollapse(false);

  return (
    <Checkbox
      id="consent"
      label="I agree to the terms and conditions"
      isRequired
      details={
        <>
          <Text size="small" marginBottom="space-0">
            Please review our terms and conditions before you agree.
          </Text>

>           <Link elementType="button" color="inherit" underlined="always" onClick={toggleHandler} aria-expanded={isOpen}>

Reviewer: ```suggestion
          <Link elementType="button" color="inherit" underlined="always" onClick={toggleHandler} aria-expanded={isOpen} aria-controls="checkbox-terms-collapse">
```

Reviewer: It would be useful to better document these aria attributes in the Collapse/README.md.
